code/parachain/frame/assets-registry/src/lib.rs

```diff
@@ -187,6 +212,10 @@ pub mod pallet {
 		/// `ed` - same meaning as in for foreign asset account (if None, then asset is not
 		/// sufficient)
 		#[pallet::weight(<T as Config>::WeightInfo::register_asset())]
+		// TODO(RFC-0013): Update Assets Registry - Pallet Functions
+		// | Update `register_*_asset` routes
```

Comment:
Local and foreign need not be separate, they both have a universal asset identifier in MultiAsset/MultiLocation. You register once with full path you can derive if they are local or not with code.

Reply:
I was not originally planning to derive an XCM Multi-Location for each asset, only those who need it.
If I'm not mistaken, we'll need to decide on a way to derive locations for assets in our consensus system.